src/__tests__/index.test.ts

```diff
@@ -1,131 +1,420 @@
+
+const ETH_CAIP2_CHAINID = 'eip155:1';
+
+const erc721Contract = '0x7e789e2dd1340971de0a9bca35b14ac0939aa330';
+const erc721Owner = '0x431cf61e7aff8e68142f6263e9fadde40aff8c7d';
+const erc721OwnerResponse = { data: { tokens: [ { owner: { id: erc721Owner } } ] } };
+const erc721NoResponse = { data: { tokens: [ ] } };
+
+const erc1155Contract = '0x06eb48572a2ef9a3b230d69ca731330793b65bdc';
+const erc1155Owners = [ '0xef1bd3fc679a6f0cd296b59aff99ddc21409869f', '0x5bb822302e78c978f3d73cd7565ad92240779cae', '0xa84de981f6f6d2d78e8d59239db73c89f058cb58' ];
+const erc1155Accounts = erc1155Owners.map(owner => { return { account: { id: owner } } } );
+const erc1155OwnersResponse = { data: { tokens: [ { balances: erc1155Accounts } ] } };
+const erc1155NoResponse = { data: { tokens: [ { balances: [ ] } ] } };
+
+const blockQueryNumber = '1234567';
+const blockQueryResponse = { data: { blocks: [ { number: blockQueryNumber } ] } };
+
+const caipLinkControllerDid = 'did:3:testing';
+
+enum ErcNamespace {
+  ERC721 = 'erc721',
+  ERC1155 = 'erc1155'
 }
 
-describe('3ID DID Resolver', () => {
-  const chainid = 'eip155:1337'
-  let config
-  let account1, account2
-  let did1, did2, did3
+describe('NFT DID Resolver (TheGraph)', () => {
+  let config: NftResovlerConfig;
+  let nftResolver: ResolverRegistry;
+  let resolver: Resolver;
+
+  let ethAccount: string;
+  let ethAuthProv: EthereumAuthProvider;
+
 
   beforeAll(async () => {
     config = {
-      ceramic: new Ceramic(),
-      ethereumRpcs: {
-        [chainid]: 'http://localhost:8545'
-      }
-    }
-    const provider = new ethers.providers.JsonRpcProvider('http://localhost:8545')
-    const signer1 = provider.getSigner(1)
-    account1 = (await signer1.getAddress()).toLowerCase()
-    const signer2 = provider.getSigner(2)
-    account2 = (await signer2.getAddress()).toLowerCase()
-    //const sign = await signer.signMessage('asdfasdf')
-    //console.log('p', sign)
-    const factory = new ContractFactory(erc721Abi, erc721Bytecode, signer1)
-    const contract = await factory.deploy()
-    // mint some tokens
-    const a = await contract.awardItem(account1, 'ceramic://lolol1')
-    const b = await contract.awardItem(account2, 'ceramic://lolol2')
-    did1 = makeDID(chainid, contract.address, 1)
-    did2 = makeDID(chainid, contract.address, 2)
-    did3 = makeDID(chainid, contract.address, 3)
-    // create caip10-link
-
-    const ethauthprov = new EthereumAuthProvider({
-      send: async (data, cb) => {
-        if (data.method === 'eth_chainId') {
-          cb(null, {result: '0x0539'})
-        } else if (data.method === 'eth_getCode') {
-          cb(null, {result: '0x'})
-        } else {
-          // it's personal_sign
-          const msg = u8a.toString(u8a.fromString(data.params[0].slice(2), 'base16'))
-          const sign = await signer2.signMessage(msg)
-          cb(null, {result: sign})
+      ceramic: new Ceramic()
```

Comment:
This should not be needed anymore. You could just access ceramic from the global scope. No need to use the http client.

Reply:
Yup that certainly made the tests faster :sweat_smile: